routes/trackRouter.js

```diff
@@ -1074,34 +1074,34 @@ router.get('/', function (req, res, next) {
                 if (req.app.locals.main.result) {
                     return res.render("track", req.app.locals.main.result);
                 } else {
-                    // TODO: at this point, user's credentials are valid, but they do not have custom curated tracks
-                    //  display an 'error' page, which tells user they need to take a photo, and has a button prompting to go back to homepage
-                    return res.render('error', {
-                        message: "No custom tracks yet, please take a photo to get curated tracks",
-                        error: {status: 400, stack: "no stack"}
-                    });
+                    return res.redirect("/webcam/#/error/no-photo-submitted");
                 }
             } else {
                 return res.redirect("/spotify/login");
             }
         })
         .catch((err) => {
-            // TODO: just in case checkCredentials throws an error (highly unlikely)
-            console.log(err);
+            return res.redirect("/spotify/login");
         })
 });
 
 // REQUIRES: req.body to contain a list of track URI's in format ["spotify:track:1ue7zm5TVVvmoQV8lK6K2H", ...]
 router.post('/', function (req, res, next) {
     let main = req.app.locals.main;
-    return main.createMoodPlaylist(req.body)
-        .then((playlistURL) => {
-            return res.status(200).json({link: playlistURL});
-        })
-        .catch((err) => {
-            console.log(err);
-            return res.status(400).json({"error": err.message});
-        })
+
+    if(main) {
+        return main.createMoodPlaylist(req.body)
+            .then((playlistURL) => {
+                return res.status(200).json({link: playlistURL});
+            })
+            .catch((err) => {
+                return res.status(500).json({errorMsg: "Please try again later </br> </br>" + err.message});
```

Comment:
NOTE FOR US BOTH:
- errors that are thrown without promise.reject will skip this catch block
- err.message will not work for errors without an error message (need to make sure all errors thrown by our code and from external APIs have messages

Reply:
I think all errors thrown have a message by default, not too sure though

Reply:
Hmm, I can try to do a more robust check but I think some external errors may not have an error msg

Reply:
Yee lmk if you find any external errors that don't have it

Reply:
I havent run into so far but I'm thinking of reading the docs for Spotify and FaceAPI just to make sure- do you have any other ideas on how to trigger or check if they all have a message?

Reply:
For Spotify, you could try messing up some of the parameters, like if the limit is 100 u can try 110 and that should break it I think, or just any improperly formatted input.

For AzureFaceApi, same idea you can give it improperly formatted input, so attach no dataURL to the request body i think

Reply:
Reviewed their docs and played around with it- we should be good!